contracts/DecentralizedKV.sol

```diff
@@ -122,7 +122,13 @@ contract DecentralizedKV {
         bytes32 skey = keccak256(abi.encode(msg.sender, key));
         PhyAddr memory paddr = kvMap[skey];
         require(paddr.hash != 0, "data not exist");
-        require(paddr.kvSize >= off + len, "beyond the range of kvSize");
+        if (decodeType == DecodeType.PaddingPer31Bytes) {
+            // ksSize is file size
```

Comment:
ksSize => kvSize; also the comment should be "kvSize is the actual data size that dApp contract stores"